### ISSUE TITLE
[Bug]: Migrating ownCloud 10.12.1 to NextCloud 25.0.6 - all app token vanished

### ISSUE THREAD

**Issue by AlphaDE** [nextcloud/server#38007]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I migrated my ownCloud installation (due to the lacking php8 support) to Nextcloud and followed the documentation under https://docs.nextcloud.com/server/latest/admin_manual/maintenance/migrating_owncloud.html (migrating from 10.12.1 to 25.0.6).

After the otherwise sucessful migration, all app tokens were lost. Using many of them to allow apps on different platforms/devices to access data to an 2FA protected account, this caused a lot of work to create and setup them.

### Steps to reproduce

1. Have a current owncloud server
2. Download Nextcloud 25.0.6
3. Follow migration procedure

### Expected behavior

App tokens should be converted as passwords were.

### Installation method

Community Web installer on a VPS or web space

### Nextcloud Server version

25

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 7.4

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated to a major version (ex. 22.2.3 to 23.0.1)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "25.0.6.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "mail_smtpmode": "php",
        "maintenance": false,
        "theme": "",
        "loglevel": 3,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "integrity.check.disabled": true,
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379,
            "timeout": 0,
            "password": "***REMOVED SENSITIVE VALUE***",
            "dbindex": 0
        },
        "filelocking.enabled": true,
        "files_external_allow_create_new_local": "true",
        "versions_retention_obligation": "disabled",
        "memcached_servers": [
            [
                "localhost",
                11211
            ]
        ],
        "memcached_options": {
            "14": 50,
            "15": 50,
            "19": 50,
            "20": 50,
            "8": 50,
            "-1001": true,
            "16": true,
            "18": true
        },
        "files_antivirus.av_cmd_options": "",
        "files_antivirus.av_path": "\/usr\/bin\/clamscan",
        "allow_user_to_change_mail_address": "",
        "encryption.legacy_format_support": false,
        "encryption.key_storage_migrated": false,
        "default_phone_region": "DE",
        "mysql.utf8mb4": true
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.17.0
  - bruteforcesettings: 2.5.0
  - calendar: 4.3.3
  - circles: 25.0.0
  - cloud_federation_api: 1.8.0
  - comments: 1.15.0
  - contacts: 5.2.0
  - contactsinteraction: 1.6.0
  - dashboard: 7.5.0
  - dav: 1.24.0
  - federatedfilesharing: 1.15.0
  - federation: 1.15.0
  - files: 1.20.1
  - files_antivirus: 5.0.0
  - files_external: 1.17.0
  - files_pdfviewer: 2.6.0
  - files_rightclick: 1.4.0
  - files_sharing: 1.17.0
  - files_texteditor: 2.15.0
  - files_trashbin: 1.15.0
  - files_versions: 1.18.0
  - firstrunwizard: 2.14.0
  - logreader: 2.10.0
  - lookup_server_connector: 1.13.0
  - nextcloud_announcements: 1.14.0
  - notifications: 2.13.1
  - oauth2: 1.13.0
  - password_policy: 1.15.0
  - photos: 2.0.1
  - privacy: 1.9.0
  - provisioning_api: 1.15.0
  - recommendations: 1.4.0
  - related_resources: 1.0.4
  - serverinfo: 1.15.0
  - settings: 1.7.0
  - sharebymail: 1.15.0
  - support: 1.8.0
  - survey_client: 1.13.0
  - systemtags: 1.15.0
  - tasks: 0.14.5
  - text: 3.6.0
  - theming: 2.0.1
  - twofactor_backupcodes: 1.14.0
  - twofactor_totp: 7.0.0
  - updatenotification: 1.15.0
  - user_status: 1.5.0
  - viewer: 1.9.0
  - weather_status: 1.5.0
  - workflowengine: 2.7.0
Disabled:
  - admin_audit
  - encryption: 2.13.0
  - suspicious_login
  - user_ldap
```

### Nextcloud Signing status

```shell
Disabled
```

**Comment by maederm** [comment]
Had the same issue migrating from owncloud 10.12.1.3 to nextcloud 25.0.7

**Comment by joshtrichards** [comment]
> all app tokens were lost

By this I assume you mean they aren't visible in the UI? No errors in the logs when you load the app password settings page?

Any chance the instance `secret` was changed / lost during the migration? It's stored in the config. 

IIRC v25 was also around the time we were doing some cleanup for handling older installations that had empty `secret` values so maybe an odd interaction there.

Are the old tokens still visible in the database (in the `PREFIX_authtokens` table)? (Well at this point you'd probably have to check an old database snapshot or something).

**Comment by nextcloud-command** [comment]
This issue has been automatically marked as stale because it has not had recent activity and seems to be missing some essential information. It will be closed if no further activity occurs. Thank you for your contributions.

**Comment by joshtrichards** [comment]
Possibly addressed by #49523

**Comment by maederm** [comment]
@joshtrichards : At least with version 30.0.5 the v2 entries are still ignored and also not listed in the user account
